dockerfiles/bloodhound.Dockerfile

```diff
@@ -88,27 +46,83 @@ RUN wget \
   https://github.com/SpecterOps/AzureHound/releases/download/${AZUREHOUND_VERSION}/AzureHound_${AZUREHOUND_VERSION}_windows_arm64.zip \
-  https://github.com/SpecterOps/AzureHound/releases/download/${AZUREHOUND_VERSION}/AzureHound_${AZUREHOUND_VERSION}_windows_arm64.zip.sha256
+  https://github.com/SpecterOps/AzureHound/releases/download/${AZUREHOUND_VERSION}/AzureHound_${AZUREHOUND_VERSION}_windows_arm64.zip.sha256 \
+  /tmp/azurehound
+
+WORKDIR /tmp/azurehound
 RUN sha256sum -cw *.sha256
 RUN 7z x '*.zip' -oartifacts/*
-RUN ls
 
 WORKDIR /tmp/azurehound/artifacts
 RUN 7z a -tzip -mx9 azurehound-${AZUREHOUND_VERSION}.zip *
 RUN sha256sum azurehound-${AZUREHOUND_VERSION}.zip > azurehound-${AZUREHOUND_VERSION}.zip.sha256
 
+########
+# UI Build
+################
+FROM --platform=$BUILDPLATFORM docker.io/library/node:22-alpine3.20 AS ui-builder
+
+WORKDIR /build
+COPY --parents constraints.pro package.json **/package.json yarn* .yarn*  ./
+RUN yarn install
+
+COPY --parents cmd/ui packages/javascript ./
+RUN yarn build
+
+########
+# Version Build
+################
+FROM --platform=$BUILDPLATFORM docker.io/library/golang:1.23-alpine3.20 AS ldflag-builder
+ENV VERSION_PKG="github.com/specterops/bloodhound/src/version"
+RUN apk add --update --no-cache git
+WORKDIR /build
+COPY .git ./.git
+
+# sort by semver version to grab latest and convert to required ldflags
+# (see https://git-scm.com/docs/git-config#Documentation/git-config.txt-versionsortsuffix)
+RUN git --no-pager -c 'versionsort.suffix=-rc' tag --list v*.*.* --sort=-v:refname | head -n 1 | sed 's/^v//' | awk \
+  -F'[.+-]' \
+  -v pkg="$VERSION_PKG" \
+  '{ major = $1; minor = $2; patch = $3; pre = ""; if ($4) pre = $4; \
+    printf("-X '\''%s.majorVersion=%s'\'' ", pkg, major); \
+    printf("-X '\''%s.minorVersion=%s'\'' ", pkg, minor); \
+    printf("-X '\''%s.patchVersion=%s'\''", pkg, patch); \
+    if (pre != "") \
+      printf(" -X '\''%s.prereleaseVersion=%s'\''", pkg, pre); \
+  }' > LDFLAGS
+
+########
+# API Build
+################
+FROM --platform=$BUILDPLATFORM docker.io/library/golang:1.23-alpine3.20 AS api-builder
+
+ARG TARGETOS
+ARG TARGETARCH
+ENV CGO_ENABLED=0
+ENV GOOS=$TARGETOS
+ENV GOARCH=$TARGETARCH
+
+RUN apk add --update --no-cache git
+RUN mkdir -p /opt/bloodhound /etc/bloodhound /var/log
+
+WORKDIR /build
+COPY --parents go* cmd/api packages/go ./
+COPY --from=ldflag-builder /build/LDFLAGS ./
+COPY --from=ui-builder /build/cmd/ui/dist ./cmd/api/src/api/static/assets
+RUN --mount=type=cache,target=/go/pkg/mod go build -C cmd/api/src -o /bloodhound -ldflags "$(cat LDFLAGS)" github.com/specterops/bloodhound/src/cmd/bhapi
+
 ########
 # Package Bloodhound
 ################
 FROM gcr.io/distroless/static-debian11 AS bloodhound
 ARG SHARPHOUND_VERSION
 ARG AZUREHOUND_VERSION
 
+COPY --from=api-builder /bloodhound /opt/bloodhound /etc/bloodhound /var/log /
```

Comment:
Do we know why we copy `/var/log`? It seems a bit odd, though it's been that way for a while from the looks

Reply:
My guess is it's an artifact from a long time ago when we were building a debian package for enterprise. I'd remove it but I'm not 100% sure we won't break something.

Reply:
Yeah, that's fair. I'm not really sure what could possibly rely on it, but we'd need a good way to test that it doesn't break anything